### ISSUE TITLE
Maven publishing incorrectly fails build due to duplicate coordinates when artifactId is set.

### ISSUE THREAD

**Issue by anuraaga** [gradle/gradle#12176]
### Expected Behavior

When there are two projects with the same project name but `MavenPublication` has  `artifactId` set manually with different values, publishing should be allowed as they are not duplicates.

### Current Behavior

Build fails with error message like

```
> Project :proj2:framework has the same (groupId, artifactId) as :proj1:framework. You should set both the groupId and artifactId of the publication or opt out by adding the org.gradle.dependency.duplicate.project.detection system property to 'false'.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
```

### Context

I have two projects with the same name at different paths which should work with all functionality. Naturally, Maven coordinates need to be unique, which they are through use of `artifactId` and `archivesBaseName`.

It seems that a change was made to fix circular dependencies when projects have the same names. This is great news! But it seems to have introduced behavior to also check for duplicate coordinates when publishing. The logic seems wrong to me though, as it only checks `Project.group` and `Project.name` though what matters more are `MavenPublication.groupId` and `MavenPublication.artifactId` (which default to the project values but can be overridden by users).

https://github.com/gradle/gradle/blob/master/subprojects/maven/src/main/java/org/gradle/api/publish/maven/plugins/MavenPublishPlugin.java#L293

### Steps to Reproduce 

```
git clone https://github.com/anuraaga/gradle-circular-maven
cd gradle-circular-maven
./gradlew :proj2:framework:publishToMavenLocal
```

Notice the maven coordinates are `test.proj1-framework` and `test.proj2-framework` by manually specifying `artifactId`. But this is being treated as a duplicate and fails to publish.

### Your Environment

Build scan URL: https://gradle.com/s/zxf4zjhgga3f4

**Comment by ljacomet** [comment]
Fixing the underlying issue of duplicate project names has some corner cases where the new detection mechanism gets triggered in a valid setup.

You have two options, as indicated by this sentence in the error message:
> You should set both the groupId and artifactId of the publication or opt out by adding the org.gradle.dependency.duplicate.project.detection system property to 'false'.

* Set the `groupId` as well on the publication, something like `groupId = project.group`
* Add the system property to `gradle.properties` in your build

However it looks like we missed this as a potential breaking change, so it will be added to the relevant documentation section for the 6.2 release.

**Comment by ljacomet** [comment]
Note has been added to the upgrade guide

**Comment by anuraaga** [comment]
I understand the warning message / note provide a good workaround but shouldn't we keep this open as a bug in the current logic? Gradle can generate unique poms even when the false positive triggers - I think this means it's clear the fully evaluated build is enough information to do a precise duplicate check. The check could be moved to a later stage when it's more precise, probably when running pom generation task, can't it?

**Comment by ljacomet** [comment]
The logic introduced as a fix for #847 has been reverted in Gradle 6.2.1
